src/main/java/org/apache/commons/collections4/bloomfilter/HasherCollection.java

```diff
@@ -106,6 +120,23 @@ public boolean forEachIndex(IntPredicate consumer) {
         };
     }
 
+    /**
+     * Creates an IndexProducer of comprising the unique indices across all the contained
+     * hashers.
+     *
+     * <p>This is equivalent to an IndexProducer created from a Bloom filter that comprises all
+     * the contained hashers.</p>
+     *
+     * @param shape the shape of the desired Bloom filter.
+     * @return the iterator of integers
+     */
+    public IndexProducer absoluteUniqueIndices(final Shape shape) {
+        return consumer -> {
+            Objects.requireNonNull(consumer, "consumer");
+            return uniqueIndices(shape).forEachIndex(IndexFilter.create(shape, consumer));
```

Comment:
IIUC this can exceed the capacity of the IndexFilter. The uniqueIndices(shape) will output up to k indices for each hasher in the collection (eliminating duplicates). Thus if we have two hashers with shape(k=5, n=1000) we could output 10 indices from the combined unique indices. Then we create an IndexFilter for shape which could be optimised only to check up to k=5 indices.

This throws an index out of bounds exception from IndexFilter.ArrayTracker.

```Java
@Test
public void testAbsoluteUniqueIndices() {
    int[] actual = new HasherCollection(
        new IncrementingHasher(1, 1),
        new IncrementingHasher(10, 1)
    ).absoluteUniqueIndices(Shape.fromKM(5, 1000)).asIndexArray();
    int[] expected = IntStream.concat(
            IntStream.range(1, 1 + 5),
            IntStream.range(10, 10 + 5)
        ).toArray();
    Assertions.assertArrayEquals(expected, actual);
}
```

When I update HasherCollection absoluteUniqueIndices to use a fake shape that forces the IndexTracker.BitMapTracker implementation then the above tests passes:
```Java
public IndexProducer absoluteUniqueIndices(final Shape shape) {
    return consumer -> {
        Objects.requireNonNull(consumer, "consumer");
        return uniqueIndices(shape).forEachIndex(IndexFilter.create(
                Shape.fromKM(shape.getNumberOfBits(), shape.getNumberOfBits()), consumer));
    };
}
```

I think that this can be easily solved by having a new package-private factory constructor in IndexFilter to choose the implementation that can handle exceeding the number of hash functions in the shape:

```Java
static IntPredicate create(int numberOfBits, IntPredicate consumer) {
    // Choose the BitMapTracker implementation to handle any number of hash functions
```

Or you could create a new Shape as:
```Java
public IndexProducer absoluteUniqueIndices(final Shape shape) {
    return consumer -> {
        Objects.requireNonNull(consumer, "consumer");
        return uniqueIndices(shape).forEachIndex(IndexFilter.create(
                Shape.fromKM(shape.getNumberOfHashFunctions() * hashers.size(),
                             shape.getNumberOfBits()), consumer));
    };
}
```
Perhaps the later is simpler.